source/client/process_impl.cc

```diff
@@ -270,10 +270,31 @@ ProcessImpl::mergeWorkerStatistics(const std::vector<ClientWorkerPtr>& workers)
   return merged_statistics;
 }
 
+void ProcessImpl::allowApiV2(envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+  auto* admin_layer = bootstrap.mutable_layered_runtime()->add_layers();
+  admin_layer->set_name("admin layer");
+  admin_layer->mutable_admin_layer();
+  envoy::config::bootstrap::v3::RuntimeLayer* runtime_layer =
+      bootstrap.mutable_layered_runtime()->add_layers();
+  runtime_layer->set_name("static_layer");
+  Envoy::ProtobufWkt::Value proto_true;
+  proto_true.set_string_value(std::string("true"));
+  (*runtime_layer->mutable_static_layer()
+        ->mutable_fields())[std::string("envoy.reloadable_features.enable_deprecated_v2_api")] =
+      proto_true;
+  (*runtime_layer->mutable_static_layer()
+        ->mutable_fields())[std::string("envoy.features.enable_all_deprecated_features")] =
```

Comment:
Just for context, do we really need to enable all deprecated features to enable v2 api?  That seems much more far reaching.

If so, I wonder if we should instead create a generic functionality to enable deprecated features, rather than specifically enabling v2 api.

Reply:
Good catch, we don't need to enable all deprecated features, we can target just what we need (allowing usage of the deprecated v2 api). Done.